cmd/tap-device-maker/tap-device-maker.go

```diff
@@ -0,0 +1,75 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/songgao/water"
+)
+
+func createTapDevice(name string, isMultiqueue bool) error {
+	var err error = nil
+	config := water.Config{
```

Comment:
Would it be possible to do this with vishwandanda/netlink? I did not look it up, so can be perfectly reasonable to use this library.

Reply:
I've provided a draft implementation of this in https://github.com/kubevirt/kubevirt/pull/3290/commits/54460a1c7084f487b18cc789abd8596ce55cb961 .

It depends on what we want to do; currently it does not work for multi-queue (the netlink API is different: it needs the user to indicate the number of queues to use, so if we choose to have this, I'll need to expose the multi-queue logic within the converter as a function that can be re-used when requesting the tap device to be created).

Reply:
Thanks for the effort. the additional library is fine fine for me to use.